newlib/libc/sys/vita/lcltime_r.c

```diff
@@ -0,0 +1,51 @@
+/*
+
+Copyright (C) 2019, Martin Larralde
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+
+*/
+
+#include <errno.h>
+#include <sys/time.h>
+#include <sys/types.h>
+
+#include <psp2/rtc.h>
+
+#include "vitaerror.h"
+
+#define TRY(x) {rid=x; if (rid < 0) {errno = __vita_sce_errno_to_errno(rid); return NULL;}}
```

Comment:
maybe we need to update `error.c` file or just use `errno = rid`

https://github.com/vitasdk/vita-headers/blob/bbaaf16/include/psp2/rtc.h#L17-L31

Reply:
`lcltime_r` is not supposed to error with anything else than `EOVERFLOW`, but most of the RTC errors would probably resolve as `EINVAL` I guess ? Anyway I went with `errno = rid` for now.